### PR TITLE
PARQUET-234: Add ParquetInputSplit methods for compatibility.

### PR DESCRIPTION
**Pull request by rdblue**

### CODE REVIEW

**Comment by julienledem** [comment]
What's the expected benefit of returning dummy values like empty collections or null?
If something calls those methods, it will fail or worse, have unexpected behavior.
If we can not reproduce the original behavior, it should either throw an UnsupportedOperationException or not be reinstated.

In short we should decide what should happen when those methods are called before reintroducing them.

File: parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputSplit.java [hunk 1]
```diff
@@ -158,6 +168,24 @@ public ParquetInputSplit(
...
+    return null;
+  }
+
+  /**
+   * @return the file schema
+   * @deprecated the file footer is no longer read before creating input splits
+   */
+  @Deprecated
+  public String getFileSchema() {
+    return null;
```
Comment (rdblue) [inline parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputSplit.java]: TODO: throw UnsupportedOperationException.

File: parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputSplit.java [hunk 2]
```diff
@@ -115,6 +116,15 @@ private static long end(List<BlockMetaData> blocks, String requestedSchema) {
   }
 
   /**
+   * @return the block meta data
+   * @deprecated the file footer is no longer read before creating input splits
+   */
+  @Deprecated
+  public List<BlockMetaData> getBlocks() {
+    return Collections.emptyList();
```
Comment (rdblue) [inline parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputSplit.java]: TODO: throw UnsupportedOperationException

**Comment by rdblue** [comment]
@julienledem, I think you're right that some should throw instead of returning fake values. But for some, I think the empty values work. For example, returning a null requested schema should just cause readers to use the file schema. And I think it should be fine for the metadata accessors to return empty (but valid) metadata. That way there's a chance that older code will succeed.

**Comment by rdblue** [comment]
@julienledem, I added UnsupportedOperationException for all of the methods, as you suggested. I think you're right about it and we can't fake any of that information. We can only make sure that programs that didn't access it at all will succeed, which is what this commit does by making the class binary compatible but throwing exceptions for any behavior changes.

File: parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputSplit.java [hunk 3]
```diff
@@ -26,6 +26,7 @@
 import java.io.DataOutputStream;
 import java.io.IOException;
 import java.util.Arrays;
+import java.util.Collections;
```
Comment (julienledem) [inline parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputSplit.java]: not used?

**Comment by julienledem** [comment]
looks good to me.
+1

**Pull request closed without merging**